modules/adpod.js

```diff
@@ -0,0 +1,356 @@
+/**
+ * This module houses the functionality to evaluate and process adpod adunits/bids.  Specifically there are several hooked functions,
+ * that either supplement the base function (ie to check something additional unique to adpod objects) or to replace the base funtion
+ * entirely when appropriate.
+ *
+ * Brief outline of each hook:
+ * - `callPrebidCacheHook` - for any adpod bids, this function will temporarily hold them in a queue in order to send the bids to Prebid Cache in bulk
+ * - `checkAdUnitSetupHook` - evaluates the adUnits to ensure that required fields for adpod adUnits are present.  Invalid adpod adUntis are removed from the array.
+ * - `checkVideoBidSetupHook` - evaluates the adpod bid returned from an adaptor/bidder to ensure required fields are populated; also initializes duration bucket field.
+ *
+ * To initialize the module, there is an `initAdpodHooks()` function that should be imported and executed by a corresponding `...AdServerVideo`
+ * module that designed to support adpod video type ads.  This import process allows this module to effectively act as a sub-module.
+ */
+
+import * as utils from '../src/utils';
+import { addBidToAuction, doCallbacksIfTimedout, AUCTION_IN_PROGRESS } from '../src/auction';
+import { store } from '../src/videoCache';
+import { hooks } from '../src/hook';
+import { config } from '../src/config';
+import { ADPOD } from '../src/mediaTypes';
+import Set from 'core-js/library/fn/set';
+import find from 'core-js/library/fn/array/find';
+const from = require('core-js/library/fn/array/from');
+
+export const TARGETING_KEY_PB_CAT_DUR = 'hb_pb_cat_dur';
+export const TARGETING_KEY_CACHE_ID = 'hb_cache_id'
+
+let queueTimeDelay = 50;
```

Comment:
any experimentation with different default values here?

Reply:
I did test using different values (went to 500) and with the timers I setup at different points in this module - it was executing the call to Prebid Cache at the set time value when there was a sufficient gap in bids.